### PR TITLE
Fixing emulator bug #391

### PR DESCRIPTION
**Pull request by lukolszewski**
The emulator would crash for me. This is either because SDL was not initialised before SDL_WaitEvent was called, or more likely because anothe thread was trying to use SDL at the same time. This fixes both by initialising SDL earlier and by setting up a mutex that locks access to SDL when the input thread works. 

**This affects the emulator only, it has no impact on running on the goggles**

### CODE REVIEW

File: src/core/input_device.c [hunk 1]
```diff
@@ -14,6 +14,7 @@
 #include <minIni.h>
 
 #ifdef EMULATOR_BUILD
+#include "SDLaccess.h"
 #include <SDL2/SDL.h>
```
Comment (SumolX) [inline src/core/input_device.c:18]: Don't need to include this header since its already included via sdlaccess.h

File: src/core/main.c [hunk 2]
```diff
@@ -11,6 +11,12 @@
 #include <lvgl/lvgl.h>
 #include <minIni.h>
 
+#ifdef EMULATOR_BUILD
+#include "SDLaccess.h"
+#include <SDL2/SDL.h>
```
Comment (SumolX) [inline src/core/main.c:16]: Same issue here.  No need to include SDL.h here either.

File: src/core/main.c [hunk 3]
```diff
@@ -141,6 +147,13 @@ void lvgl_init() {
 int main(int argc, char *argv[]) {
     pthread_mutex_init(&lvgl_mutex, NULL);
 
+#ifdef EMULATOR_BUILD
+    global_sdl_mutex = SDL_CreateMutex();
+    if (global_sdl_mutex == NULL) {
+        // Handle error: SDL_CreateMutex failed
```
Comment (SumolX) [inline src/core/main.c:153]: Please add a log entry.  Good to know if it ever fails.

File: src/ui/ui_porting.c [hunk 4]
```diff
@@ -10,6 +10,7 @@
 #include "lvgl/lvgl.h"
 
 #ifdef EMULATOR_BUILD
+#include "SDLaccess.h"
 #include <SDL2/SDL.h>
```
Comment (SumolX) [inline src/ui/ui_porting.c:14]: Please remove SDL.h here as well.

File: src/ui/ui_porting.c [hunk 5]
```diff
@@ -127,8 +130,11 @@ int lvgl_init_porting() {
 
     lv_disp_drv_init(&disp_drv);
 #else
-    SDL_InitSubSystem(SDL_INIT_VIDEO);
+    if (SDL_WasInit(SDL_INIT_VIDEO) == 0) {
+        SDL_InitSubSystem(SDL_INIT_VIDEO);
+    }
```
Comment (SumolX) [inline src/ui/ui_porting.c:135]: maybe add else { log it (not that it should ever happen) }

File: src/core/input_device.c (hunk 1 above)
Comment (lukolszewski) [reply src/core/input_device.c]: done

File: src/core/main.c (hunk 2 above)
Comment (lukolszewski) [reply src/core/main.c]: done

File: src/ui/ui_porting.c (hunk 4 above)
Comment (lukolszewski) [reply src/ui/ui_porting.c]: done

File: src/ui/ui_porting.c (hunk 5 above)
Comment (lukolszewski) [reply src/ui/ui_porting.c:136]: Added. It may happen if the input thread managed to initialise first.

**Comment by lukolszewski** [comment]
Thanks for the review, all changes were added..

File: src/core/main.c (hunk 3 above)
Comment (lukolszewski) [reply src/core/main.c]: done

**Comment by ligenxxxx** [comment]
@lukolszewski Build firmware failed. Please fix this.

**Comment by SumolX** [comment]
@lukolszewski Is the PR abandoned?  If not please do update the branch and we can get it merged into main.  Thank You.

**Comment by eliasrenman** [comment]
I went ahead and fixed this over at #392 so please have a look there 😀

**Pull request merged**